### PR TITLE
Version flag

### PR DESCRIPTION
**Pull request by jmccormick2001**
**Description of the change:**
This PR adds a version flag to the operator-sdk scorecard subcommand which will support
the ability of the end user to specify which version of the scorecard they want to run.

With this flag added, typical usage in code looks like:

        if schelpers.IsV1alpha2() {
                // v1alpha2 logic
        }

The version command option is allowed on the scorecard CLI command line or
within the scorecard configuration file.

If not set, the version value is v1alpha1 which represents the current scorecard implementation.

This PR includes code to validate the version:  v1alpha1, v1alpha2 being the valid choices

When a user specifies an invalid version an error is reported as follows:

        $ operator-sdk scorecard --version foomatic
        Error: invalid scorecard version (foomatic); valid values: v1alpha1, v1alpha2
        Usage:
          operator-sdk scorecard [flags]

This PR provides the version implementation within the schelpers package can can imported within the scorecard and scplugins packages when needed.

The PR includes a unit test which tests the version validation logic.

**Motivation for the change:**

This PR supports a future scorecard implementation and allows the scorecard to support different implementations by switching upon the requested version.

### CODE REVIEW

File: internal/pkg/scorecard/helpers/versions.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+package schelpers
+
+import (
+	"fmt"
+	"github.com/spf13/viper"
+	"strings"
+)
+
+const V1alpha1 = "v1alpha1"
```
Comment (estroz) [inline internal/pkg/scorecard/helpers/versions.go]: These don't need to be exported, unless you pan on using them outside of this package.

File: internal/pkg/scorecard/helpers/versions.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+const LatestScorecardVersion = V1alpha2
+const VersionOpt = "version"
+
+var ScorecardVersions = []string{DefaultScorecardVersion, LatestScorecardVersion}
+
+func ValidateVersion(version string) error {
+	validVersion := false
+	for _, a := range ScorecardVersions {
+		if a == version {
+			validVersion = true
```
Comment (estroz) [inline internal/pkg/scorecard/helpers/versions.go]: You can just `return nil` here, and `return fmt.Errorf("invalid ...")` at the function's end.

File: internal/pkg/scorecard/helpers/versions.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import (
+	"fmt"
+	"github.com/spf13/viper"
+	"strings"
+)
+
+const V1alpha1 = "v1alpha1"
+const V1alpha2 = "v1alpha2"
+const DefaultScorecardVersion = V1alpha1
```
Comment (estroz) [inline internal/pkg/scorecard/helpers/versions.go]: I like this default/latest abstraction.

File: internal/pkg/scorecard/helpers/versions.go (hunk 1 above)
Comment (jmccormick2001) [reply internal/pkg/scorecard/helpers/versions.go]: thanks, will fix this and resubmit the PR.

File: internal/pkg/scorecard/helpers/versions.go (hunk 2 above)
Comment (jmccormick2001) [reply internal/pkg/scorecard/helpers/versions.go]: thanks, good catch.

File: internal/pkg/scorecard/helpers/versions.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		if a == version {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid scorecard version (%s); valid values: %s", version, strings.Join(ScorecardVersions, ", "))
+
+}
+
+func IsV1alpha2() bool {
+	if viper.Sub("scorecard").GetString(VersionOpt) == v1alpha2 {
```
Comment (camilamacedo86) [inline internal/pkg/scorecard/helpers/versions.go]: Is it checking the versions of the projects API?
If yes, should not these validations make with Regex? Have we checked if the Kubebuilder is validating the version?
Could we not add it to not allow the users to create the API'sbythe tool with an invalid value for versions as well?

Comment (jmccormick2001) [reply internal/pkg/scorecard/helpers/versions.go]: this is checking the scorecard version only, and providing a validation of what scorecard version the user enters on the command line or within the scorecard configuration file.

**Comment by camilamacedo86** [comment]
If the goal here is to create a flag in order to keep 2 implementations of the scorecard ( the old one and the new one ) then IMHO --latest or --new would be enough, do not you agree? Also, should not better change the current implementation instead of creating more than one version? if we have the old and the new one then we will need to keep maintained both.  

c/c @estroz

**Comment by jmccormick2001** [comment]
@camilamacedo86 The ability to pin scorecard to a particular version (v1alpha2) is what I was going for here, this way people can migrate to or try out the new output version if they want to and when they want to.  The default is still v1alpha1.  At some point, people might want this to default to the latest (e.g. v1alpha2), but I would consider that a future PR to make that default change.

I'm just starting to work on the actual implementation of the v1alpha2,  the goal is to have a single scorecard implementation as today, but switch in the current code when required to achieve the v1alpha2 output format.

File: cmd/operator-sdk/scorecard/cmd.go [hunk 5]
```diff
@@ -36,12 +37,14 @@ func NewCmd() *cobra.Command {
 	scorecardCmd.Flags().String(scorecard.ConfigOpt, "", fmt.Sprintf("config file (default is '<project_dir>/%s'; the config file's extension and format can be .yaml, .json, or .toml)", scorecard.DefaultConfigFile))
 	scorecardCmd.Flags().String(scplugins.KubeconfigOpt, "", "Path to kubeconfig of custom resource created in cluster")
 	scorecardCmd.Flags().StringP(scorecard.OutputFormatOpt, "o", scorecard.TextOutputFormat, fmt.Sprintf("Output format for results. Valid values: %s, %s", scorecard.TextOutputFormat, scorecard.JSONOutputFormat))
+	scorecardCmd.Flags().String(schelpers.VersionOpt, schelpers.DefaultScorecardVersion, fmt.Sprintf("scorecard version (experimental version is '%s'", schelpers.LatestScorecardVersion))
```
Comment (camilamacedo86) [inline cmd/operator-sdk/scorecard/cmd.go]: I understand that nomenclature for us is would be tech preview.

File: internal/pkg/scorecard/helpers/versions.go (hunk 4 above)
Comment (camilamacedo86) [inline internal/pkg/scorecard/helpers/versions.go]:
```suggestion
	if viper.Sub("scorecard").GetString(VersionOpt) == LatestScorecardVersion {
```

**Comment by camilamacedo86** [comment]
HI @jmccormick2001,

Really tks for the clarifications. If we will move forward with so shows just missing add a new flag in the docs and geat job 👍 by the way.

File: internal/pkg/scorecard/helpers/versions.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	for _, a := range ScorecardVersions {
+		if a == version {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid scorecard version (%s); valid values: %s", version, strings.Join(ScorecardVersions, ", "))
+
+}
+
+func IsV1alpha2() bool {
```
Comment (estroz) [inline internal/pkg/scorecard/helpers/versions.go]: I assume this will be used for something in another PR?

Comment (jmccormick2001) [reply internal/pkg/scorecard/helpers/versions.go]: yep, I was planning on using that next up to switch with.
